src/plugin/plugin.ts

```diff
@@ -72,21 +72,25 @@ export const createPlugin = <
       );
 
       newState.doc.descendants((node, pos) => {
+        // Update nodes representing any elements that are no longer selected.
         if (isProseMirrorElementSelected(node) && !elementNodeToPos.get(node)) {
-          // Update nodes representing any elements that are no longer selected.
           tr.setNodeMarkup(pos, undefined, {
```

Comment:
One alternative might be to do:
```Typescript
if(isProseMirrorElementSelected(node)){
  const newAttrs = {
      ...node.attrs,
      [elementSelectedNodeAttr]: elementNodeToPos.get(node) === undefined,
  };
  
  tr.setNodeMarkup(pos, undefined, newAttrs);
  
  return false;
}
```

You could then remove the second if block.

Reply:
Discussed with @jonathonherbert, this approach causes all PME elements to re-render. There are more efficient ways to only update what's necessary. See: https://github.com/guardian/prosemirror-elements/pull/183#issuecomment-1025905153